src/Validator/Constraints/PhoneNumber.php

```diff
@@ -54,12 +54,6 @@ class PhoneNumber extends Constraint
      */
     public function __construct($format = null, $type = null, string $defaultRegion = null, string $regionPath = null, string $message = null, array $groups = null, $payload = null, array $options = [])
     {
-        if (\is_array($format)) {
-            $options = array_merge($format, $options);
-        } elseif (null !== $format) {
```

Comment:
The first `if` seems to be (unfortunately) an important BC break. We are going to keep it besides it's _not great_.

The $format var is therefore not `string|array|null` but `int|array|null`.

I really think the `$options['value']` part is to drop because the `getDefaultOption` is not defined.

We should trigger a deprecation if an option array is passed instead of the format.

Finally we should add some doc to the docbloc to make the usage more clear and maybe rename the format `$formatOrOptions`.